### PR TITLE
DOC: add concatenation-section to text.rst

### PR DESCRIPTION
**Pull request by h-vetinari**
As requested by @jreback in #20347.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. very minor comment

**Comment by jreback** [comment]
cc @jorisvandenbossche @TomAugspurger

File: doc/source/text.rst [hunk 1]
```diff
@@ -199,6 +199,75 @@ regular expression object will raise a ``ValueError``.
...
+    s.str.cat(t, na_rep='-')
+
+Series are *not* aligned on their index before concatenation:
+
+.. ipython:: python
+
+    u = pd.Series(['b', 'd', 'a', 'c'], index=[1, 3, 0, 2])
+    # without alignment
+    s.str.cat(u)
+    # manual alignment
```
[the remark this answers is not in the document]
Comment (h-vetinari) [reply doc/source/text.rst]: Here's another try. As an aside, the lack of alignment *isn't* user-friendly (which is exactly the point of #20347)

**Comment by jreback** [comment]
thanks

**Pull request merged**